fetch_system_config/debian/fetch-melodic-config.postinst

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+set -e
+
+case "$1" in
+    "configure")
+        # Add ROS user, give access to audio/hardware
+        if ! getent passwd ros >/dev/null; then
+          adduser --disabled-password --quiet --system \
+            --no-create-home --home=/var/lib/ros --shell /bin/bash \
+            --gecos "ROS system user" --group ros
+        fi
+        adduser --quiet ros audio
+        adduser --quiet ros dialout
+        adduser --quiet ros plugdev
+        if ! getent group ros > /dev/null 2>&1; then
+            addgroup --system ros --quiet
+        fi
+        if [ "$(id -gn ros)" = "nogroup" ]; then
+            usermod -g ros ros
+        fi
+
+        # Setup /var/lib/ros as home directory
+        if [ ! -e "/var/lib/ros" ]; then
+            mkdir /var/lib/ros
+        fi
+        chown -R ros:ros /var/lib/ros
+            chmod 2775 /var/lib/ros
+
+        # Setup /var/log/ros for logging
+        if [ ! -e "/var/log/ros" ]; then
+            mkdir /var/log/ros
+        fi
+        chown -R ros:ros /var/log/ros
+        chmod 2775 /var/log/ros
+
+        # Copy the robot.launch if not already existent
+        if [ ! -e "/etc/ros/melodic/robot.launch" ]; then
+            mkdir -p /etc/ros/melodic
+            cp /opt/ros/melodic/share/fetch_bringup/launch/fetch.launch /etc/ros/melodic/robot.launch
+        fi
+
+        # TODO this file is missing in 18.04?
+        # Override /etc/acpi/powerbtn.sh
+        echo "/sbin/shutdown -h now \"Power button pressed\"" > /etc/acpi/powerbtn.sh
+        echo "exit 0" >> /etc/acpi/powerbtn.sh
+
+        # Do not wait for GRUB (we're headless)
+        sed -i "s/GRUB_TIMEOUT=.*\"\"/GRUB_TIMEOUT=10/" /etc/default/grub
+        echo "GRUB_RECORDFAIL_TIMEOUT=\$GRUB_TIMEOUT" >> /etc/default/grub
+
+        # Disable apport
+        if [ -e "/etc/default/apport" ]; then
+            sed -i "s/enabled=1/enabled=0/g" /etc/default/apport
+        fi
+
+        # Run logrotation cron job hourly
+        if [ ! -e "/etc/cron.hourly/logrotate" ]; then
+            cp /etc/cron.daily/logrotate /etc/cron.hourly/logrotate
+        fi
+
+        # Probalby null-op; We replace the update-rc.d with upstart job so that sixad starts properly
+        update-rc.d -f sixad remove
+
+        ### Set up ethernet to ensure static network for internal robot communications.
+        # Reboot required for changes to take effect.
+        # Modify default/grub to not use new-style network device naming
+        sed -i "s/GRUB_CMDLINE_LINUX=\"\"/GRUB_CMDLINE_LINUX=\"net.ifnames=0 biosdevname=0\"/" /etc/default/grub
+        update-grub
+        # Determine which device to map to eth0 and eth1 and update /etc/udev/rules.d/70-persistent-net.rules
```

Comment:
It is for our robots because I found the computer hardware we use is not 100% guaranteed to have the same e.g. ensp1s0 style names every time it boots. Therefore I implement a "standard" way to disable this newer naming behavior.  See the commit message as well.  I'll also being including notes about this networking decision in the docs for 14.04->18.04.